### PR TITLE
Flag Presets

### PR DESCRIPTION
**Pull request by Reszolute**
| Changelog |
[+] Fixed the blacklist. (Can't delete default presets now) 
[+] Added a confirmation when deleting custom flag presets. 
[+] Added icons.

Preview:
https://github.com/Skylumz/Sollumz/assets/113695236/d2576752-73ce-4aa8-b875-6e2c4438a4d3

### CODE REVIEW

**Comment by ook3D** [comment]
Balls

File: ybn/operators.py [hunk 1]
```diff
@@ -328,7 +328,26 @@ class SOLLUMZ_OT_delete_flag_preset(SOLLUMZ_OT_base, bpy.types.Operator):
...
+        "Stair mesh",
+        "Deep surface",
+        ]
+
+    def invoke(self, context, event):
+        self.confirm_delete = False
+        wm = context.window_manager
+        return wm.invoke_props_dialog(self)
+    
+    confirm_delete = bpy.props.BoolProperty(default=False)
```
Comment (alexguirre) [inline ybn/operators.py:346]: Unused property, remove it.

File: ybn/operators.py [hunk 2]
```diff
@@ -328,7 +328,26 @@ class SOLLUMZ_OT_delete_flag_preset(SOLLUMZ_OT_base, bpy.types.Operator):
...
-    preset_blacklist = ["Default"]
+    preset_blacklist = [
+        "General (Default)",
+        "General 2",
+        "Water surface",
+        "Leaves - Bush",
+        "Stair plane",
+        "Stair mesh",
+        "Deep surface",
+        ]
```
Comment (alexguirre) [inline ybn/operators.py:339]: Unindent the `]` 4 spaces.

**Review by alexguirre — CHANGES_REQUESTED** [review]
When deleting a blacklisted preset, it still shows the confirmation dialog before showing the "Cannot delete" message. Should just show the message directly.

**Review by alexguirre — APPROVED** [review]

**Pull request closed without merging**